api/migrations/20200224195202_change-state-to-inhouse.js

```diff
@@ -0,0 +1,39 @@
+exports.up = async knex =>
+  knex('apds')
+    .select('document', 'id')
+    .then(apds =>
+      Promise.all(
+        apds.map(({ document, id }) =>
+          knex('apds')
+            .where('id', id)
+            .update({
+              document: {
+                ...document,
+                activities: document.activities.map(
+                  ({ quarterlyFFP, ...activity }) => ({
+                    quarterlyFFP: Object.entries(quarterlyFFP).reduce(
+                      (quarterlyFFPObj, [ffy, quarters]) => ({
+                        ...quarterlyFFPObj,
+                        [ffy]: Object.entries(quarters).reduce(
+                          (ffyObj, [quarter, { contractors, state }]) => ({
+                            ...ffyObj,
+                            [quarter]: {
+                              contractors,
+                              inHouse: state
+                            }
+                          }),
+                          {}
+                        )
+                      }),
+                      {}
+                    ),
+                    ...activity
+                  })
+                )
+              }
+            })
```

Comment:
Yes, this is kind of ugly. I am toying with building up a list of diffs instead and using the `jsonpatch` module to apply them, since we already have it anyway, but I'm pretty sure that would also be ugly and wouldn't use "plain" Javascript.